src/LibraryManager.Contracts/FileHelpers.cs

```diff
@@ -245,21 +253,33 @@ public static bool DeleteFiles(IEnumerable<string> filePaths)
         /// Deletes folders if empty
         /// </summary>
         /// <param name="folderPaths"></param>
+        /// <param name="rootDirectory"></param>
         /// <returns></returns>
-        public static bool DeleteEmptyFoldersFromDisk(IEnumerable<string> folderPaths)
+        public static bool DeleteEmptyFoldersFromDisk(IEnumerable<string> folderPaths, string rootDirectory)
         {
+            if (folderPaths.Count() == 0)
+            {
+                return true;
+            }
+
+            HashSet<string> newFolderPaths = new HashSet<string>();
+
             try
             {
                 foreach (string path in folderPaths)
                 {
-                    DirectoryInfo directory = new DirectoryInfo(path);
-                    if (directory.Exists && IsDirectoryEmpty(path))
+                    if (!path.Equals(rootDirectory, StringComparison.CurrentCultureIgnoreCase))
                     {
-                        directory.Delete(true);
+                        DirectoryInfo directory = new DirectoryInfo(path);
```

Comment:
I don't see a need to create a DirectoryInfo object here.  We've been using the File/Directory static methods elsewhere.  This segment could be rewritten as:

```
if (Directory.Exists(path) && Directory.GetFileSystemEntries(path).Any())
{
    Directory.Delete(path);
    newFolderPaths.Add(Path.GetDirectoryName(path));
}
```
(This also gets rid of the need for IsDirectoryEmpty from this method, I'm not sure if it's needed elsewhere).